webapp-django/crashstats/crashstats/models.py

```diff
@@ -609,93 +630,72 @@ class TelemetryCrash(SocorroMiddleware):
     aliases = {"crash_id": "uuid"}
 
 
+class PermissionsReducerError(Exception):
+    pass
+
+
+@functools.lru_cache
+def get_processed_crash_permissions_reducer(permissions_tuple):
+    """Build a reducer with the schema for specified permissions
+
+    :arg permissions_tuple: tuple of permissions to reduce by
+
+    """
+    if "public" not in permissions_tuple:
+        raise PermissionsReducerError(
+            f"public not in permissions_tuple {permissions_tuple}"
+        )
+
+    permissioned_schema = transform_schema(
+        schema=PROCESSED_CRASH_SCHEMA,
+        transform_function=(
+            permissions_transform_function(permissions_have=permissions_tuple)
+        ),
+    )
+
+    return SocorroDataReducer(schema=permissioned_schema)
+
+
 class ProcessedCrash(SocorroMiddleware):
+    # FIXME(willkg): change this to BotoS3CrashStorage
     implementation = SimplifiedCrashData
```

Comment:
At some point, I want to get rid of `SimplifiedCrashData` which seems like it just adds another abstraction layer without a lot of value.